CMakeLists.txt

```diff
@@ -146,6 +146,50 @@ elseif (TARGET_PLATFORM STREQUAL "linux")
     set(BEEROCKS_BH_WIRE_IFACE "eth0")
 endif()
 
+# Test support
+option(BUILD_TESTS "build multiap unit tests" OFF)
+if (BUILD_TESTS)
```

Comment:
I wouldn't make unit tests an option. They should be mandatory

Reply:
@RanRegev wrote:
> I am uncertain about using only one variable for all kind of tests in the system.

@mariomaz wrote:
> I wouldn't make unit tests an option. They should be mandatory

Google Test may not be available - especially when cross-compiling. Generally, the unit tests are only useful for developers, not (or not much) for integrators or users. So it should definitely be possible to build without tests. Anything more fine-grained than enable or disable is probably not necessary.

We could consider making it automatic based on the availability of Google Test. However, my buildroot experience tells me that it is very convenient when integrating that you can just disable all the tests - then you don't need to be concerned with a test that fails to build on some obscure architecture. Since the BUILD_TESTS option already exists, I wouldn't remove it.

Reply:
> Since the BUILD_TESTS option already exists, I wouldn't remove it.

Ok. 

> Google Test may not be available - especially when cross-compiling

GoogleTest *should* be available even when cross-compiling (we have the sources and the CMakeFiles required to cross-compile). Unit tests should be written with the same care and rules as regular code, so they *should* also cross-compile.

It's not usual at all, but I've seen cases where unit tests passed in the development environment but failed in a particular target platform. And, after debugging, we discovered that the problem was actually in the code under test, not in the test itself.  Cross-compiled tests paid it off.

Admittedly, compiling and running unit tests in *all* possible target platforms is more expensive and we can decide it's not worth it for some platform. Thus having the possibility to disable building the tests is handy.

Reply:
I would vote for as many tests as possible, all the time.
It is very easy to decide not to write/run tests, but ad Mario mentioned - it pays off big time.